app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class ReportsController < ApplicationController
+  before_action :find_own_report, only: %i[edit update destroy]
+  before_action :find_report, only: [:show]
+
+  # GET /reports or /reports.json
+  def index
+    @reports = Report.all
+  end
+
+  # GET /reports/1 or /reports/1.json
+  def show
+    @report = Report.find(params[:id])
+  end
+
+  # GET /reports/new
+  def new
+    @report = Report.new
+  end
+
+  # GET /reports/1/edit
+  def edit; end
+
+  # POST /reports or /reports.json
+  def create
+    @report = current_user.reports.build(report_params)
+
+    if @report.save
+      redirect_to report_url(@report), notice: t('controllers.common.notice_create', name: Report.model_name.human)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /reports/1 or /reports/1.json
+  def update
+    if @report.update(report_params)
+      redirect_to report_url(@report), notice: t('controllers.common.notice_update', name: Report.model_name.human)
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /reports/1 or /reports/1.json
+  def destroy
+    @report.destroy
+    redirect_to reports_url, notice: t('controllers.common.notice_destroy', name: Report.model_name.human)
+  end
+
+  private
+
+  def find_own_report
```

Comment:
```suggestion
  def set_own_report
```

インスタンス変数に値をセットする系のメソッドは `set_` で始めることが多いです

Reply:
そうなのですね🤔
インスタンス変数に値をセットするメソッドは`set_`で名付けること、承知しました🙏